### DISCUSSION TITLE
Solar Passthrough Questions

### DISCUSSION THREAD

**Discussion by stanpete78** [hoylabs/OpenDTU-OnBattery#358] [Q&A]
Hey there,
so I've read the other discussion about solar passthrough but it seems I'm to ... slow ... to understand.

My preferred solution would be to use as much of the solar energy right away and only store the excess energy in the Pylontech battery. And, going further (maybe at night or during cloudy weather), have the DTU spend the stored energy in a "smart" way.
I'm no battery expert and having the wrong settings makes me a bit nervous, regarding the battery's life expectancy - 

so would someone be so nice to suggest a some values for SOC Start/Stop/Passtrough? 
Experimenting with mine, I'm at 80/15/15 at the moment.

thanks!
stan

**Comment by schlimmchen** [comment]
Well, that's no easy question. I understand that it is hard to get your head around all the settings and how they actually work. It took me longer than I am willing to admit, too. However, it's best you understand what the settings do and chose them yourself, rather than listening to somebody else.

These are my five cents:
* Why are you choosing a SoC start threshold of 80? Are you sure you want to charge the battery so aggressively? It is a valid choice, but why do you want to keep a charged battery?
* Whats the third value, the second "15"? *Full* solar passthrough threshold? If so, than that's definitely wrong. *Full* solar passthrough is not "regular" solar passthrough. You only want your inverter output to be matched with the solar charger output if the battery is already full. All solar energy is fed into the grid in full solar passthrough mode.
* My start threshold is 20%, the stop threshold is 15%, and the full solar passthrough threshold is 90%.
* How far you are willing to charge and discharge your battery is solely your call. And might depend on the BMS in use. Less Depth of Discharge (DoD) might increase your battery lifetime. But that's a tradeoff between DoD and capacity.

**Reply by stanpete78** [reply]
Oh, wow, that was quick.

I "chose" a start value of 80 because it was set as default :-)
The 15% for full solar passthrough was, yes, because of the confusion between those "two" different passthroughs. I wasn't aware of this even after reading the descriptions.

Still it seems you brought me on the right track. I'm setting the start value down to 20% as recommended and will watch it doing its thing for a bit.

So thanks a lot for now, my friend.
st

**Comment by stanpete78** [comment]
So after watching the system on a rainy day like today, I now developed a bit more sense for how it‘s behaving. There is a thing I noticed, that might make it a bit smarter though.

So today for example, there was not that much sunshine so the SOC only climbed slowly to my new start threshold of 20%. At 20% the inverter started and the house pulled the entire 800W that I had set as a maximum. So the battery was quickly discharged until it reached the stop threshold of 10%. Then, with the inverter off, it climbed to 20% again, and so on and so on.

So my question would be: 
Wouldn‘t it be good to have a third threshold in between the other two that sets a different maximum power for the inverter?
For example: In the morning I start with 10% and it climbs up to 20%, the inverter turns on, the battery gets depleted - but then, at 12%, a value of let‘s say 50W is now the maximum the inverter is allowed to use. So it will most likely not reach 10% and therefore not stop the inverter and therefore save less energy in the battery and therefore waste less energy in the process.

I really like to know what you guys think about this.

st

**Reply by peff74** [reply]
I would support that proposal.

**Reply by spcqike** [reply]
i'm not quite sure but isn't that the normal behaviour if you use solar-passthrough? like shown [here](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/367)

as long as there is solar production, the inverter is limited to said production. it shouldn't drain the battery at all. if available, excess energy will be stored in the battery. 
at night, or without production (clouds), battery will be used, if you are within the thresholds.

**Reply by stanpete78** [reply]
It may be because I haven‘t connected my MPPT to the DTU. Everything seems a bit off and that could be the reason, right?

Only problem is: I have two MPPT controllers. Would connecting only one make a difference?

In Home Assistant I have a good overwiew of what Is beeing generated by all the solar panels just by combining the values battery_current, battery_voltage and hoymiles (output) power. If OpenDTU-onBattery would just use those numbers, it might not be neccessary to connect via ve.direct and you could use more than one MPPT controller.
